dex/msgjson/types.go

```diff
@@ -845,14 +845,25 @@ type OrderBook struct {
 	QuoteFeeRate uint64           `json:"quoteFeeRate"`
 }
 
+type BasicMatch struct {
+	MatchID  Bytes  `json:"matchid"`
+	MakerOID Bytes  `json:"maker"`
+	TakerOID Bytes  `json:"taker"`
```

Comment:
> @chappjc: My suggestion to include the OIDs in the BasicMatch was really just for the server->client messaging, which gives the backend some more capability to audit the matches based on the match_proof and other orderbook subscription data.

Can you be more specific?

This is > 200 bytes **per match** sent to every consumer. I think we'd need a pretty compelling reason to justify that.

Reply:
The `MatchID` itself can be omitted, but what I'm describing is how an observer can determine what orders were matched in an epoch.  This can only be done at present by:
- subscribe to orderbook
- observe all the epoch order notes in a full epoch
- at epoch close use the match_proof note with the preimages listed to collect the orders that should be matched
- **run the matching algorithm** against book - major burden, not for the casual user

The idea of just sending the matches (this PR) is to avoid the above if we just want matches for display purposes.  If someone really wants to audit the matches, they can go to the trouble the steps above with an orderbook subscription.

Reply:
I don't agree that a casual user needs that much detail, but I'm just one guy. How about we just do a slice of tuples? 

```Go
Matches [][2]uint64 `json:"matches"` // [rate, qty]
```

and I guess we should send it in the `EpochReportNote`. Is it relevant to the match proof?

Reply:
That's actually what I'm suggesting in my other comment https://github.com/decred/dcrdex/pull/1663#issuecomment-1199709874 Except I think we can aggregate all matches for a given rate and sum the quantity.

I do concede that every match in an epoch isn't quite necessary, but I do think that giving the amounts and rates, even if aggregated over many matches, is important.  No exchange I have ever seen does not give you this info.

Reply:
So, I think we can remove `BasicMatch`, and add 

```
	// quantity of matches by rate: {rate}qty
	Matches map[uint64]uint64 `json:"matches"`
```

into `MatchCycleStats` so it is reported on `EpochReportNote`?

Reply:
That seems reasonable.  Using a map to aggregate quantities for a rate is certainly convenient (e.g. `ms[rate]+=qty` just works even before the first entry for a given `rate`).

As for the JSON in the epoch report, a slice of tuples might end up better for unmarshalling.

```json
"matchSummary": [
  [11000, 20],
  [11400, 40],
  [12000, 60],
]
```

vs.

```json
"matchSummary": {
  "11000": 20,
  "11400": 40,
  "12000": 60,
}
```

Because when you marshal a `map[uint64]uint64`, it makes the key into a string.  But as long as the quoted key represents a number it unmarshals fine, so maybe it's fine.  https://go.dev/play/p/uJO6JL64CuV  Go ahead and feel it out.

Reply:
I pushed the sugested changes, aggregating on a map [rate]quantity. 

Still need to fix the site build, to show this information properly, but pushed it before changing the front, so I can get your opinions.

Reply:
The problem with a tuple would be that we would need to run the array in order to find the rate so we can sum new matches from the server, while with a map we would simple sum it up, like you said: `ms[rate]+=qty`. 

So I believe it would be simpler to just convert the matchSummary from string to uint, but I am fine with either way. 

Which one do you prefer @chappjc ?

Reply:
> The problem with a tuple would be that we would need to run the array in order to find the rate so we can sum new matches from the server, while with a map we would simple sum it up, like you said: `ms[rate]+=qty`.

I guess I have to review your current code, but I thought that aggregation happens only on the server to reduce an epochs matches, and we can send anything else to the client although it might require converting and sorting I suppose.

> So I believe it would be simpler to just convert the matchSummary from string to uint, but I am fine with either way.
> 
> Which one do you prefer @chappjc ?

No preference yet.  I'd have to get into the code and feel it out.  I don't feel like it's a critical design decision except that updating a msgjson type commits us to live with that choice.